src/main/kotlin/net/irisshaders/lilybot/extensions/util/ModUtilities.kt

```diff
@@ -110,6 +116,95 @@ class ModUtilities : Extension() {
 			}
 		}
 
+		/**
+		 * Message editing command
+		 *
+		 * @since 3.3.0
+		 */
+		ephemeralSlashCommand(::SayEditMessageArgs) {
+			name = "edit-say-message"
+			description = "Edit a message created by /say"
+
+			check {
+				anyGuild()
+				configPresent()
+				hasPermission(Permission.ModerateMembers)
+			}
+
+			action {
+				// The channel the message was sent in. Either the channel provided, or if null, the channel the
+				// command was executed in.
+				val channelOfMessage = if (arguments.channelOfMessage != null) {
```

Comment:
Can you not use an elvis operator here? `val channelOfMessage = (guild!!.getchannel(arguments.channelofmessage?.id) as MessageChannelBehavior) ?: channel`

Reply:
No that wouldn't work. We're checking the argument not if the channel is null

Reply:
From what I can tell there is no proper way to make that work. Plus something with such a large string of variables passing through it would probably benefit from clearer code layout